harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -184,6 +195,33 @@ def download(self, path: Optional[str] = None) -> str:
 
         return str(local_ckpt_dir)
 
+    def _download_via_master(self, local_ckpt_dir: pathlib.Path) -> None:
+        local_ckpt_dir.mkdir(parents=True, exist_ok=True)
+        if os.name == "nt":
+            fext = "zip"
+            mime_type = "application/zip"
+        else:
+            fext = "tgz"
+            mime_type = "application/gzip"
+        resp = self._session.get(f"/checkpoints/{self.uuid}", headers={"Accept": mime_type})
+        if not resp.ok:
+            raise RuntimeError(
+                "unable to download checkpoint from master:", resp.status_code, resp.reason
+            )
+
+        tmpfile = pathlib.Path(f"{local_ckpt_dir}/{self.uuid}.{fext}")
+        with open(tmpfile, "wb") as f:
+            f.write(resp.content)
+
+        try:
+            if fext == "zip":
+                subprocess.run(["unzip", "-d", local_ckpt_dir, tmpfile], check=True)
```

Comment:
Let's not do this as a subprocess (or deal with os-specific codepaths).

Instead, let's use the [tar library](https://docs.python.org/3/library/tarfile.html) that comes with python in all cases, and do it in python.

Reply:
Sure, and [zipfile](https://docs.python.org/3/library/zipfile.html) for Windows?